tachyon/zk/plonk/circuit/selector_compressor.h

```diff
@@ -55,76 +56,129 @@ class SelectorCompressor {
-  static Result Process(const std::vector<std::vector<bool>>& selectors_in,
-                        const std::vector<size_t>& degrees, size_t max_degree,
-                        AllocateFixedColumnCallback callback) {
-    if (selectors_in.empty()) return {};
+  void Process(const std::vector<std::vector<bool>>& selectors_in,
+               const std::vector<size_t>& degrees, size_t max_degree,
+               AllocateFixedColumnCallback callback) {
+    if (selectors_in.empty()) return;
+
+    callback_ = std::move(callback);
 
+    // For example, suppose we have the following inputs:
+    //
+    // |selectors_in| = [
+    //   [1, 0, 0, 0, 0, 0, 0, 0, 1],
+    //   [1, 0, 0, 0, 0, 0, 0, 1, 0],
+    //   [1, 0, 0, 0, 0, 0, 1, 0, 0],
+    //   [0, 1, 0, 0, 0, 1, 1, 1, 0],
+    //   [0, 1, 0, 0, 1, 0, 1, 0, 1],
+    //   [0, 1, 0, 1, 0, 0, 0, 1, 1],
+    //   [0, 0, 1, 1, 1, 0, 0, 0, 0],
+    //   [0, 0, 1, 1, 0, 1, 0, 0, 0],
+    //   [0, 0, 1, 0, 1, 1, 0, 0, 0],
+    //   [1, 1, 1, 1, 1, 1, 1, 1, 1],
+    // ]
+    //
+    // |degrees| = [3, 3, 3, 3, 3, 3, 7, 7, 7, 0]
+    //
+    // |max_degree| = 10
+    //
     // The length of all provided selectors must be the same.
     size_t n = selectors_in[0].size();
     DCHECK(std::all_of(selectors_in.begin(), selectors_in.end(),
                        [n](const std::vector<bool>& activations) {
                          return activations.size() == n;
                        }));
 
-    std::vector<SelectorDescription> selectors =
+    // |selectors| = [
+    //   s₀: SelectorDescription(0, [1, 0, 0, 0, 0, 0, 0, 0, 1], 3),
+    //   s₁: SelectorDescription(1, [1, 0, 0, 0, 0, 0, 0, 1, 0], 3),
+    //   s₂: SelectorDescription(2, [1, 0, 0, 0, 0, 0, 1, 0, 0], 3),
+    //   s₃: SelectorDescription(3, [0, 1, 0, 0, 0, 1, 1, 1, 0], 3),
+    //   s₄: SelectorDescription(4, [0, 1, 0, 0, 1, 0, 1, 0, 1], 3),
+    //   s₅: SelectorDescription(5, [0, 1, 0, 1, 0, 0, 0, 1, 1], 3),
+    //   s₆: SelectorDescription(6, [0, 0, 1, 1, 1, 0, 0, 0, 0], 7),
+    //   s₇: SelectorDescription(7, [0, 0, 1, 1, 0, 1, 0, 0, 0], 7),
+    //   s₈: SelectorDescription(8, [0, 0, 1, 0, 1, 1, 0, 0, 0], 7),
+    //   s₉: SelectorDescription(9, [1, 1, 1, 1, 1, 1, 1, 1, 1], 0),
+    // ]
+    selectors_ =
         base::Map(selectors_in,
                   [&degrees](size_t i, const std::vector<bool>& activations) {
                     size_t max_degree = degrees[i];
                     return SelectorDescription(i, &activations, max_degree);
                   });
 
-    std::vector<std::vector<F>> combination_assignments;
-    std::vector<SelectorAssignment<F>> selector_assignments;
+    HandleZeroDegreeSelectors();
 
+    // All of the remaining |selectors| are simple. Let's try to combine them.
+    // First, we compute the exclusion matrix.
+    // See tachyon/zk/plonk/circuit/exclusion_matrix.h for details.
+    ExclusionMatrix exclusion_matrix(selectors_);
+
+    // combination for s₀, s₃, s₆:
```

Comment:
```suggestion
    // Then, we combine the remaining |selectors|.
    // combination for s₀, s₃, s₆:
```
There is nothing after "First, we compute ~", so I thought it would be better to add a comment for readability.